src/StatsTools/variational_energy_estimator.jl

```diff
@@ -0,0 +1,79 @@
+"""
+    variational_energy_estimator(shifts, overlaps; skip=0, kwargs...)
+    variational_energy_estimator(df::DataFrame; skip=0, kwargs...)
+Compute the variational energy estimator from the replica time series of the `shifts` and
+coefficient vector `overlaps` by blocking analysis.
+Keyword arguments are passed on to [`ratio_of_means()`](@ref).
+Returns a [`RatioBlockingResult`](@ref).
+
+An estimator for the variational energy
+```math
+\\frac{⟨\\mathbf{c}⟩^† \\mathbf{H}⟨\\mathbf{c}⟩}{⟨\\mathbf{c}⟩^†⟨\\mathbf{c}⟩}
+```
+is calculated from
+```math
+Ē_{v}  =  \\frac{\\sum_{a<b}^R \\overline{(S_a+S_b) \\mathbf{c}_a^† \\mathbf{c}_b}}
+               {2\\sum_{a<b}^R \\overline{\\mathbf{c}_a^† \\mathbf{c}_b}} ,
+```
+where the sum goes over distinct pairs out of the ``R`` replicas. See
+[arXiv:2103.07800](http://arxiv.org/abs/2103.07800).
+
+The `DataFrame` version can extract the relevant information from the result of
+[`lomc!`](@ref). Set up [`lomc!`](@ref) with the keyword argument
+`replica = AllOverlaps(R)` and `R ≥ 2`.
+
+See [`AllOverlaps`](@ref).
+"""
+function variational_energy_estimator(
+    shifts::AbstractArray{<:AbstractArray},
+    overlaps::AbstractArray{<:AbstractArray};
+    kwargs...
+)
```

Comment:
```suggestion
function variational_energy_estimator(shifts, overlaps; kwargs...)
```
I would drop the type annotations here, which would allow passing tuples as well.